x-pack/plugins/siem/public/app/types.ts

```diff
@@ -38,7 +47,7 @@ export type SiemNavTabKey =
 export type SiemNavTab = Record<SiemNavTabKey, NavTab>;
 
 export interface SecuritySubPluginStore<K extends SecuritySubPluginKeyStore, T> {
-  initialState: Record<K, T>;
+  initialState: Record<K, T | undefined>;
```

Comment:
initial state should allow `undefined`. We could use `PreloadedState` helper from redux, but (esp. w/ the dynamic key) this is easier to read.